app/models/course.rb

```diff
@@ -79,12 +79,19 @@ class Course < ApplicationRecord
            through: :site_statuses
 
   has_many :enrichments,
-           ->(course) { where(provider_code: course.provider.provider_code) },
+           ->(course) {
+             where(provider_code: course.provider.provider_code)
+               .order(created_at: :asc, id: :asc)
+           },
```

Comment:
I'm not sure about ordering in the association here. Aside from the potentially trivial performance hit, this could lead to surprising results if you try to order the association. For example, suppose we wanted to grab the first draft (thinking about the code we already have), you might try this to keep things fast:

```
enrichments.order(:status).first
```

This wouldn't work as expected since AR merges `order` statements and `order(:status)` would be the primary ordering criteria and `status` would only be used where `created_at` matched. Since this will never happen, using a subsequent `order` on this association would never work.

Can we create an association that orders `enrichments`? e.g.:

```
  has_many :enrichments,
           ->(course) { where(provider_code: course.provider.provider_code) },
           foreign_key: :ucas_course_code,
           primary_key: :course_code,
           class_name: 'CourseEnrichment' do
    ...
  end

  def ordered_enrichments
    enrichments.order(created_at: :asc, id: :asc)
  end
```

Reply:
Hmm, I think the example is flawed but basis of the argument I agree with. I'll give this a go :+1: